### ISSUE TITLE
dotnet dev-certs https --clean fails with "Sequence contains no matching element"

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/DotNetTools#464]
Repro:
```
dotnet dev-certs https
dotnet dev-certs https --clean
```

Result:
```
Cleaning HTTPS development certificates from the machine. A prompt might get displayed to confirm the removal of some of the certificates.
There was an error trying to clean HTTPS development certificates on this machine.
Sequence contains no matching element
```
Details:
Windows 10 (latest)
.NET CLI 2.1.400-preview-009088

**Comment by LukeTurp** [comment]
I am experiencing precisely the same issue.

**Comment by natemcmaster** [comment]
cc @javiercn

**Comment by javiercn** [comment]
@natemcmaster I'm trying to figure what the best approach to handle these issues are as the library/tool is very conservative about throwing exceptions. There is a related issue in home about it.

**Comment by javiercn** [comment]
@LukeTurp  

I've created a powershell script to help diagnose issues with the tool. https://gist.github.com/javiercn/9f2657f88498ce1b0b8c99d326131b54

Read through the script to ensure you are ok running it, run it with .\cert-diag.ps1, evaluate the output for anything that you want to censor and paste the output here.

The script also contains a switch -TryCreateCertificate that will create a certificate equivalent to the one the tool generates using powershell. You can try and run it at your own risk and see if that fixes the issue.

**Comment by javiercn** [comment]
This issue happens when the certificate is present but not trusted. I'll take a look at it, it should be easy to fix.

**Comment by jakubsvobodacz** [comment]
Same problem here.

**Comment by javiercn** [comment]
@jac009 I believe I fixed this issue recently as part of https://github.com/aspnet/Common/blame/master/shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs#L544. 

It will be on our next preview release. In the mean time, you can use the script in the gist above to troubleshoot your configuration.

If clean is failing on windows, you can go to the certificate manager and remove the localhost certificates (they'll have a friendly name of ASPNET Core HTTPS development certificate) from the current user personal store and from the list of trusted roots for the current user.

After that, you should be able to run dotnet-dev-certs https --trust and create and trust a new certificate.

**Comment by jakubsvobodacz** [comment]
> @jac009 I believe I fixed this issue recently as part of https://github.com/aspnet/Common/blame/master/shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs#L544.
> 
> It will be on our next preview release. In the mean time, you can use the script in the gist above to troubleshoot your configuration.
> 
> If clean is failing on windows, you can go to the certificate manager and remove the localhost certificates (they'll have a friendly name of ASPNET Core HTTPS development certificate) from the current user personal store and from the list of trusted roots for the current user.
> 
> After that, you should be able to run dotnet-dev-certs https --trust and create and trust a new certificate.

I do not have any certificates with friendly name having ASPNET. Neither in Personal or Trusted.

**Comment by jakubsvobodacz** [comment]
> @jakubsvobodacz Nope. Current user.
> ![image](https://user-images.githubusercontent.com/6995051/47104721-efcab780-d1f6-11e8-934d-fe382b9b0876.png)
> 
> ![image](https://user-images.githubusercontent.com/6995051/47104788-15f05780-d1f7-11e8-9f62-f8db69cc822e.png)

Thanks a lot! I wiped them all out and tried the clean dotnet command and it stated that there is no certificate so I reproduced the steps by creating one and trust. Now it works and Kestrel is not giving me the error when debugging anymore.

**Comment by javiercn** [comment]
This should be fixed as part of 2.2 preview3
